app/system/survey/js/prompts.js

```diff
@@ -2226,6 +2226,453 @@ promptTypes.date_month_and_year_only = promptTypes.date_no_time.extend({
+        }
+    },
+
+    modification: function(evt) {
+        var that = this;
+        odkCommon.log('D',"prompts." + that.type + ".modification px: " + that.promptIdx);
+        if ( !that.insideAfterRender ) {
+            var formattedDateValue = that.$('input').val();
+            var value = that.formatDBVal(formattedDateValue);
+
+            //
+            // we are using a date pop-up.  If an earlier action fails, we should not
+            // attempt to apply the state changes of this pop-up. Tolerate the loss
+            // of whatever the user tried to pick. They shouldn't move so fast.
+
+            var ctxt = that.controller.newContext(evt, that.type + ".modification");
+            that.controller.enqueueTriggeringContext($.extend({},ctxt,{success:function() {
+
+                odkCommon.log('D',"prompts." + that.type + ".modification: determine if reRendering ", "px: " + that.promptIdx);
+                var ref = that.getValue();
+
+                var rerender = false;
+                if ( ref === null || ref === undefined ) {
+                    rerender = ( value !== null && value !== undefined );
+                } else if ( value === null || value === undefined ) {
+                    rerender = ( ref !== null && ref !== undefined );
+                } else {
+                    rerender = !(that.sameValue(ref, value));
+                }
+
+                var renderContext = that.renderContext;
+                if ( value === undefined || value === null ) {
+                    renderContext.value = '';
+                } else {
+                    renderContext.value = formattedDateValue;
+                }
+
+                // track original value
+                var originalValue = that.getValue();
+                that.setValueDeferredChange(value);
+                renderContext.invalid = !that.validateValue();
+                if ( renderContext.invalid ) {
+                    value = originalValue;
+                    formattedDateValue = moment(value).format(that.timeFormat);
+                    // restore it...
+                    that.setValueDeferredChange(originalValue);
+                    rerender = true;
+                }
+
+                renderContext.value = formattedDateValue;
+                if ( rerender ) {
+                    odkCommon.log('D',"prompts." + that.type + ".modification: reRender", "px: " + that.promptIdx);
+                    that.reRender(ctxt);
+                }  else {
+                    // We are now done with this
+                    ctxt.success();
+                }
+            },
+            failure:function(m) {
+                odkCommon.log('D',"prompts." + that.type + ".modification -- prior event terminated with an error -- aborting!", "px: " + that.promptIdx);
+                ctxt.failure(m);
+            }}));
+        }
+    },
+});
+
+promptTypes.coptic_calendar_picker = promptTypes.non_gregorian_calendar.extend({
```

Comment:
I think we can probably merge this as is, but in a future PR it would be nice to try to clean this up to not have so much duplicate code. Each of these is essentially the same except for the calendar picker swapped out.